doc/nrf/ug_app_dev.rst

```diff
@@ -0,0 +1,72 @@
+.. _ncs-app-dev:
+
+Application Development
+#######################
+
+This page describes the fundamentalas about developing a new application based
+on the |NCS|.
+
+Build and Configuration System
+******************************
+
+Zephyr's build and configuration system
+=======================================
+
+The |NCS| build and configuration system is in fact Zephyr's, which in
+turn uses the following building blocks as a foundation:
+
+  * CMake, the cross-platform build system generator
+  * Kconfig, a powerful configuration system also used in the Linux kernel
+  * Device Tree, a hardware description language that is used to describe the
+    hardware that the |NCS| will run on
+
+Since the build and configuration system used by the |NCS| is provided by
+Zephyr, we refer to the original Zephyr documentation here in order to avoid
+duplication. The relevant sections of the Zephyr documentation that cover the
+different building blocks mentioned above can be found here:
+
+  * :ref:`zephyr:application` is a complete guide to application development
+    with Zephyr, including the build an configuration system
+  * :ref:`zephyr:cmake-details` describes in-depth the usage of CMake for
+    Zephyr-based applications
+  * :ref:`zephyr:application_kconfig` contains a guide for Kconfig usage in
+    applications
+  * :ref:`zephyr:application_dt` explains how to use Device Tree and its overlays to
+    customize an application's Device Tree
+
+|NCS| additions
+===============
+
+As mentioned in the previous sections, the build and configuration system in the
+|NCS| is inherited from Zephyr. The |NCS| does however add some additional
+functionality that is covered in this section, and that you should be aware of
+when you start writing your own applications based on this SDK:
+
+  * The |NCS| provides an additional :file:`boilerplate.cmake` that must be
+    included before Zephyr's in your app's :file:`CMakeLists.txt`::
+
+      include(path/to/nrf/cmake/boilerplate.cmake)
+      include($ENV{ZEPHYR_BASE}/cmake/app/boilerplate.cmake NO_POLICY_SCOPE)
+
+    Since the |NCS| does not require or provide an environment variable
+    equivalent to :makevar:`ZEPHYR_BASE`, you can either create your own or
+    specify the full path in your own :file:`CMakeLists.txt`.
+
+  * The |NCS| build system extends Zephyr's with support for multi-image builds.
+    You can find out more about these in the :ref:`ug_multi_image` section.
+
+Custom boards
+*************
+
+Defining your own board is a very common step in application development, since
+applications are typically designed to run on boards that are not directly
+supported by the |NCS|, given that they are typically custom designs and not
+available publicly.
+In order to define your own board you can use the following Zephyr guides as
+reference, since boards are defined in the |NCS| just as they are in Zephyr:
```

Comment:
What about the OOT NCS commits that rename xxx_ns to xxx? Worth mentioning here?